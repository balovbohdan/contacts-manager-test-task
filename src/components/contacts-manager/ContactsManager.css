.main {
    max-width: 800px;
    width: calc(100% - 20px);

    margin: 30px auto;
}

.head {
    width: 100%;

    display: flex;
    justify-content: flex-end;
}

.addContactBtn {
    display: block;
    cursor: pointer;

    padding: 10px 20px;
    border-radius: var(--border-radius);
    background-color: rgba(0, 0, 0, .2);

    color: var(--color-white);
}

.addContactBtn:hover {
    transition: var(--trans);
    background-color: rgba(0, 0, 0, .3);
}