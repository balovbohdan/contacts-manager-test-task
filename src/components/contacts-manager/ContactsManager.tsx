import * as React from 'react';

import {List} from './list';
import * as T from './types';
import css from './ContactsManager.css';

export const ContactsManager = (props:T.Props) =>
    <div className={css.main}>
        <Head toggleAddContactWindow={props.toggleAddContactWindow}/>
        <List
            contacts={props.contacts}
            toggleContactWindow={props.toggleContactWindow}/>
    </div>;

const Head = ({toggleAddContactWindow}:T.HeadProps) =>
    <div className={css.head}>
        <AddContactBtn onClick={toggleAddContactWindow}/>
    </div>;

const AddContactBtn = ({onClick}:T.AddContactBtnProps) =>
    <div onClick={onClick} className={css.addContactBtn}>
        Add contact
    </div>;