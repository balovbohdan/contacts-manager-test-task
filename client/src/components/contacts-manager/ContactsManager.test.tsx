import * as React from 'react';
import {shallow} from 'enzyme';

import {ContactsManager} from './ContactsManager';

describe('ContactsManager', () => {
    const propsDef = {
        hasMoreContacts: true,
        fetchContacts: jest.fn(),
        toggleContactWindow:jest.fn(),
        toggleAddContactWindow: jest.fn(),
        contacts: [{
            id: 1,
            name: 'name',
            img: '/user.svg',
            phone: '+380967162534'
        }]
    };

    it('Should match snapshot', () => {
        const props = { ...propsDef };
        const manager = shallow(<ContactsManager {...props}/>);

        expect(manager).toMatchSnapshot();
    });
});