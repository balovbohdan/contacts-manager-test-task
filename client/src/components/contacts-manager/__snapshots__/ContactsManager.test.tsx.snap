// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsManager Should match snapshot 1`] = `
<div
  className="main"
>
  <Head
    toggleAddContactWindow={[MockFunction]}
  />
  <Component
    contacts={
      Array [
        Object {
          "id": 1,
          "img": "/user.svg",
          "name": "name",
          "phone": "+380967162534",
        },
      ]
    }
    fetchContacts={[MockFunction]}
    hasMoreContacts={true}
    toggleContactWindow={[MockFunction]}
  />
</div>
`;
